app/modules/asset/service.server.ts

```diff
@@ -101,25 +102,59 @@ type AssetWithInclude<T extends Prisma.AssetInclude | undefined> =
 export async function getAsset<T extends Prisma.AssetInclude | undefined>({
   id,
   organizationId,
+  userOrganizations,
   include,
 }: Pick<Asset, "id"> & {
   organizationId: Asset["organizationId"];
+  userOrganizations: Pick<UserOrganization, "organizationId">[];
   include?: T;
 }): Promise<AssetWithInclude<T>> {
   try {
+    const otherOrganizationIds = userOrganizations.map(
+      (org) => org.organizationId
+    );
+
     const asset = await db.asset.findFirstOrThrow({
-      where: { id, organizationId },
+      where: {
+        OR: [
+          { id, organizationId },
+          { id, organizationId: { in: otherOrganizationIds } },
+        ],
+      },
       include: { ...include },
     });
 
+    /* User is accessing the asset in the wrong organization. In that case we need special 404 handling. */
+    if (
+      asset.organizationId !== organizationId &&
+      otherOrganizationIds.includes(asset.organizationId)
+    ) {
+      throw new ShelfError({
+        cause: null,
+        title: "Asset not found",
+        message: "",
+        additionalData: {
+          type: "asset-from-other-org",
+          assetOrganization: userOrganizations.find(
```

Comment:
Ideally this already includes the `organization` data, so we can render it nicely on the error handler.

Reply:
Nevermind, i saw the query of `getUserOrganizations`, organization object is already loaded i can see.